src/teaching-element/Assessment/index.vue

```diff
@@ -166,6 +169,16 @@ export default {
       this.isValidAnswer = isValid;
     },
     submit() {
+      this.isReflection
+        ? this.submitReflection()
+        : this.submitAssessment();
```

Comment:
How about firing event from single point in code and just having two _getters_ to calculate/return its payload?
Also it would be nice to use pseudo namespace: `submit:reflection` :bowtie: